fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -744,31 +745,76 @@ protected final void setupDruidDimensionsLoader(
      * @return The cache instance
      */
     protected DataCache<?> buildCache() {
-        if (BardFeatureFlag.DRUID_CACHE_V2.isOn()) {
-            try {
-                MemTupleDataCache<String> cache = new MemTupleDataCache<>();
-                LOG.info("MemcachedClient Version 2 started {}", cache);
-                return cache;
-            } catch (IOException e) {
-                LOG.error("MemcachedClient Version 2 failed to start {}", e);
-                throw new IllegalStateException(e);
+        if (CacheFeatureFlag.TTL.isOn() || CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) { // old caching strategy
+            if (CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) {
```

Comment:
This defaulting behavior is handled in the CacheFlags below.  You can simply do a set of flag if/else's here.